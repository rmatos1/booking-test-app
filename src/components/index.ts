export * from "./bookingDescription";
export * from "./bookingForm";
export * from "./images";
export * from "./inputGroup";
export * from "./navBar";
export * from "./styledComponents";
export * from "./testComponent";
