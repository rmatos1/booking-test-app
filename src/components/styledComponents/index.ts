export * from './buttons';
export * from './formComponents';
export * from './images';
export * from './layouts';
export * from './lists';
export * from './modalComponents';
export * from './texts';
