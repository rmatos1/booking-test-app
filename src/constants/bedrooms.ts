import { images } from '../assets';

export interface IBedroom {
  number: number;
  type: string;
  beds: {
    double?: number;
    single?: number;
  };
  img: string;
  fullCapacity: number;
  dailyPrice: number;
}

/**
 * constant that stores the maximum number of people possible in the biggest bedroom
 */
export const MAX_CAPACITY_BEDROOM = 4;

/**
 * constant with the available bedrooms to be booked
 */
export const hotelBedrooms: IBedroom[] = [
  {
    number: 1,
    type: 'Quadruple',
    beds: {
      double: 1,
      single: 2,
    },
    img: images.bedroom1,
    fullCapacity: 4,
    dailyPrice: 65,
  },
  {
    number: 2,
    type: 'Single',
    beds: {
      double: 1,
    },
    img: images.bedroom2,
    fullCapacity: 2,
    dailyPrice: 40,
  },
  {
    number: 3,
    type: 'Triple',
    beds: {
      double: 1,
      single: 1,
    },
    img: images.bedroom3,
    fullCapacity: 3,
    dailyPrice: 55,
  },
  {
    number: 4,
    type: 'Triple',
    beds: {
      single: 3,
    },
    img: images.bedroom4,
    fullCapacity: 3,
    dailyPrice: 50,
  },
  {
    number: 5,
    type: 'Single',
    beds: {
      single: 2,
    },
    img: images.bedroom5,
    fullCapacity: 2,
    dailyPrice: 35,
  },
];
