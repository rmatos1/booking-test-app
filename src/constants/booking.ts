export const initialBookingData = {
  checkIn: '',
  checkOut: '',
  qtyGuests: '',
  name: '',
  email: '',
  selectedBedroom: 0,
};

export const testBookingData = {
  id: '1',
  name: 'Julia',
  email: 'julia@teste.com',
  checkIn: '2023-12-19',
  checkOut: '2023-12-26',
  selectedBedroom: 3,
  qtyGuests: 3,
  totalPrice: 450,
};
