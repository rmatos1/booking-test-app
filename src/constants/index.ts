export * from './bedrooms';
export * from './booking';
export * from './styles';
