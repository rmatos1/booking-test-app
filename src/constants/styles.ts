/*
 * break points
 */
export const SMALL_DESKTOP = '990px';
export const TABLET = '768px';
export const MOBILE = '460px';

/***************************** */

export const HORIZONTAL_PADDING_DESKTOP = '25px';

export const HORIZONTAL_PADDING_MOBILE = '15px';
