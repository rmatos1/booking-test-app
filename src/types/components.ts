import { ReactNode } from "react";

export interface IComponentWithChildren {
  children?: ReactNode;
}

export interface IBaseDrawer extends IComponentWithChildren {
  isDrawerVisible: boolean;
  onCloseDrawer: () => void;
}
