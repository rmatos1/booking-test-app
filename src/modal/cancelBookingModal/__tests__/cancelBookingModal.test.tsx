import { describe, test } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';
import moment from 'moment';
import { TestComponent } from '../../../components';
import { testBookingData } from '../../../constants';

import { CancelBookingModal } from '..';

const defaultProps = {
  onClose: jest.fn(),
  bookingData: testBookingData,
  onCancelBooking: jest.fn(),
};

const setup = (): JSX.Element => {
  return (
    <TestComponent>
      <CancelBookingModal {...defaultProps} />
    </TestComponent>
  );
};

describe('<CancelBookingModal />', () => {
  test('should render correctly', () => {
    const wrapper = render(setup());

    expect(wrapper).toMatchSnapshot();
  });

  test('should call onClose on click', () => {
    const wrapper = render(setup());

    const overlay = wrapper.getByTestId('overlay');
    fireEvent.click(overlay);

    expect(defaultProps.onClose).toHaveBeenCalled();

    const closeButton = wrapper.getByTestId('close-button');
    fireEvent.click(closeButton);

    expect(defaultProps.onClose).toHaveBeenCalled();

    const denyButton = wrapper.getByTestId('deny-button');
    fireEvent.click(denyButton);

    expect(defaultProps.onClose).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the defined check-in and check-out', () => {
    const wrapper = render(setup());

    const modalTitle = wrapper.getByTestId('cancel-modal-title');

    expect(modalTitle.textContent).toContain(
      moment(defaultProps.bookingData.checkIn).format('MM/DD/YYYY')
    );
    expect(modalTitle.textContent).toContain(
      moment(defaultProps.bookingData.checkOut).format('MM/DD/YYYY')
    );
  });

  test('should call onCancelBooking on click', () => {
    const wrapper = render(setup());

    const cancelBooking = wrapper.getByTestId('cancel-booking');
    fireEvent.click(cancelBooking);

    expect(defaultProps.onCancelBooking).toHaveBeenCalled();
  });
});
