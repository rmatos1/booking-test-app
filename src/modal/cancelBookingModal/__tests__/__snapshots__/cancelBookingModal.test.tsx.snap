// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CancelBookingModal /> should call onClose on click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bkzXAi jvfYZe"
        data-testid="cancel-booking-modal"
      >
        <div
          class="sc-idOjMB kWZquY"
          data-testid="overlay"
        />
        <div
          class="sc-dIUfKc fErPdZ"
        >
          <div
            class="sc-hHftZz dXseJl"
          >
            <button
              class="sc-dmlpXa gYemGx"
              data-testid="close-button"
            >
              X
            </button>
          </div>
          <svg
            height="80px"
            viewBox="0 0 512 512"
            width="80px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48Zm0 319.91a20 20 0 1 1 20-20 20 20 0 0 1-20 20Zm21.72-201.15-5.74 122a16 16 0 0 1-32 0l-5.74-121.94v-.05a21.74 21.74 0 1 1 43.44 0Z"
              fill="#f73936"
            />
          </svg>
          <h4
            data-testid="cancel-modal-title"
          >
            Do you want to cancel this booking from
             
            12/19/2023
             to
             
            12/26/2023
            ?
          </h4>
          <div
            class="sc-kEjbdu dgQVNr"
          >
            <button
              class="sc-jrAGKZ iDYoRl"
              data-testid="deny-button"
            >
              No
            </button>
            <button
              class="sc-jrAGKZ eAYdtz"
              data-testid="cancel-booking"
            >
              Yes
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bkzXAi jvfYZe"
      data-testid="cancel-booking-modal"
    >
      <div
        class="sc-idOjMB kWZquY"
        data-testid="overlay"
      />
      <div
        class="sc-dIUfKc fErPdZ"
      >
        <div
          class="sc-hHftZz dXseJl"
        >
          <button
            class="sc-dmlpXa gYemGx"
            data-testid="close-button"
          >
            X
          </button>
        </div>
        <svg
          height="80px"
          viewBox="0 0 512 512"
          width="80px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48Zm0 319.91a20 20 0 1 1 20-20 20 20 0 0 1-20 20Zm21.72-201.15-5.74 122a16 16 0 0 1-32 0l-5.74-121.94v-.05a21.74 21.74 0 1 1 43.44 0Z"
            fill="#f73936"
          />
        </svg>
        <h4
          data-testid="cancel-modal-title"
        >
          Do you want to cancel this booking from
           
          12/19/2023
           to
           
          12/26/2023
          ?
        </h4>
        <div
          class="sc-kEjbdu dgQVNr"
        >
          <button
            class="sc-jrAGKZ iDYoRl"
            data-testid="deny-button"
          >
            No
          </button>
          <button
            class="sc-jrAGKZ eAYdtz"
            data-testid="cancel-booking"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<CancelBookingModal /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bkzXAi jvfYZe"
        data-testid="cancel-booking-modal"
      >
        <div
          class="sc-idOjMB kWZquY"
          data-testid="overlay"
        />
        <div
          class="sc-dIUfKc fErPdZ"
        >
          <div
            class="sc-hHftZz dXseJl"
          >
            <button
              class="sc-dmlpXa gYemGx"
              data-testid="close-button"
            >
              X
            </button>
          </div>
          <svg
            height="80px"
            viewBox="0 0 512 512"
            width="80px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48Zm0 319.91a20 20 0 1 1 20-20 20 20 0 0 1-20 20Zm21.72-201.15-5.74 122a16 16 0 0 1-32 0l-5.74-121.94v-.05a21.74 21.74 0 1 1 43.44 0Z"
              fill="#f73936"
            />
          </svg>
          <h4
            data-testid="cancel-modal-title"
          >
            Do you want to cancel this booking from
             
            12/19/2023
             to
             
            12/26/2023
            ?
          </h4>
          <div
            class="sc-kEjbdu dgQVNr"
          >
            <button
              class="sc-jrAGKZ iDYoRl"
              data-testid="deny-button"
            >
              No
            </button>
            <button
              class="sc-jrAGKZ eAYdtz"
              data-testid="cancel-booking"
            >
              Yes
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bkzXAi jvfYZe"
      data-testid="cancel-booking-modal"
    >
      <div
        class="sc-idOjMB kWZquY"
        data-testid="overlay"
      />
      <div
        class="sc-dIUfKc fErPdZ"
      >
        <div
          class="sc-hHftZz dXseJl"
        >
          <button
            class="sc-dmlpXa gYemGx"
            data-testid="close-button"
          >
            X
          </button>
        </div>
        <svg
          height="80px"
          viewBox="0 0 512 512"
          width="80px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48Zm0 319.91a20 20 0 1 1 20-20 20 20 0 0 1-20 20Zm21.72-201.15-5.74 122a16 16 0 0 1-32 0l-5.74-121.94v-.05a21.74 21.74 0 1 1 43.44 0Z"
            fill="#f73936"
          />
        </svg>
        <h4
          data-testid="cancel-modal-title"
        >
          Do you want to cancel this booking from
           
          12/19/2023
           to
           
          12/26/2023
          ?
        </h4>
        <div
          class="sc-kEjbdu dgQVNr"
        >
          <button
            class="sc-jrAGKZ iDYoRl"
            data-testid="deny-button"
          >
            No
          </button>
          <button
            class="sc-jrAGKZ eAYdtz"
            data-testid="cancel-booking"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
